code/game/objects/effects/decals/cleanable/humans.dm

```diff
@@ -295,3 +295,67 @@
 	if((blood_state != BLOOD_STATE_OIL) && (blood_state != BLOOD_STATE_NOT_BLOODY))
 		return TRUE
 	return FALSE
+
+/obj/effect/decal/cleanable/blood/hitsplatter
+	name = "blood splatter"
+	pass_flags = PASSTABLE | PASSGRILLE
+	icon = 'icons/effects/blood.dmi'
+	icon_state = "hitsplatter1"
+	plane = -1
+	random_icon_states = list("hitsplatter1", "hitsplatter2", "hitsplatter3")
+	var/turf/prev_loc
+	var/mob/living/blood_source
+	var/skip = FALSE //Skip creation of blood when destroyed?
+	var/amount = 3
+
+/obj/effect/decal/cleanable/blood/hitsplatter/Initialize(mapload, blood)
+	. = ..()
+	if(blood)
+		blood_source = blood
+	prev_loc = loc //Just so we are sure prev_loc exists
+
+/obj/effect/decal/cleanable/blood/hitsplatter/proc/GoTo(turf/T, range, speed = 1)
```

Comment:
this proc looks very shitcode to me but I suppose it's not that bad, maybe I'm being dum dum